papermill/iorw.py

```diff
@@ -26,10 +28,20 @@ def __init__(self):
         self.reset()
 
     def read(self, path):
+        if not fnmatch.fnmatch(os.path.basename(path), "*.*"):
+            warnings.warn("the file is not specified with any extension : " + os.path.basename(path))
+        elif not (fnmatch.fnmatch(os.path.basename(path), "*.ipynb") or
+                  fnmatch.fnmatch(os.path.basename(path), "*.json")):
+            warnings.warn("the file extension is not in either .ipynb or .json")
```

Comment:
I would change the warning to:
```python
"The specified input file ({}) does not end in '.ipynb' or '.json'".format(path)
```

Reply:
@betatim thanks for the suggestion. Made this change :)